src/function.rs

```diff
@@ -84,3 +106,13 @@ impl FunctionRef {
         self.location
     }
 }
+
+impl<TS: TypeSystem> FunctionType<TS> {
+    pub fn can_invoke(&self) -> bool {
+        use FunctionType::*;
```

Comment:
every instance is only used once, they should just use the full name

Reply:
just expanding this to explain a bit of reasoning. The code is not much longer with the full `FunctionType::Static`, or other, names. You generally want to do it this way anyway because you typicality see the full name for enums in rust with the exception of the super common ones in the std (`Some` `None` `Err` `Ok`). We did use this technique in fender a few times because those were large match case functions and if you look at it and get confused once you understand you understand that context for the whole file; you get confused and look for what `Static` means here you only understand the context for that one line so its not worth the `*` import.